pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -470,9 +470,6 @@ func newCDIForCR(cr *hcov1alpha1.HyperConverged, namespace string) *cdiv1alpha1.
 
 func (r *ReconcileHyperConverged) ensureCDI(instance *hcov1alpha1.HyperConverged, logger logr.Logger, request reconcile.Request) error {
 	cdi := newCDIForCR(instance, UndefinedNamespace)
-	if err := controllerutil.SetControllerReference(instance, cdi, r.scheme); err != nil {
```

Comment:
line 89 in this file depends on CDI being owned by the hco.cr. With the suggested code, HCO would not be notified if someone edits CDI.cr. For completeness, please make this controller watch the cdi.cr regardless of its owner.